types/rangy/index.d.ts

```diff
@@ -1,73 +1,141 @@
+
+        getBookmark(containerNode?: Node): rangy.Bookmark;
+
+        moveToBookmark(bookmark: rangy.Bookmark): void;
+
+        getDocument(): Document;
+
+        inspect(): string;
+
+        equals(range: RangyRange): boolean;
+
+        refresh(): any;
+
+        select(): any;
+
+        toCharacterRange(containerNode: Node, opts?: any): { start: number, end: number };
+    }
 
-interface RangyStatic {
-    createNativeRange(doc?:Document|Window|HTMLIFrameElement):Range;
-    createRange(doc?:Document|Window|HTMLIFrameElement):RangyRange;
-    createRangyRange(doc?:Document|Window|HTMLIFrameElement):RangyRange;
-    getNativeSelection(win?:Window):Selection;
-    getSelection(doc?:Document|Window|HTMLIFrameElement):RangySelection;
-    addInitListener(listener:(rangy:RangyStatic) => void):any;
-    shim():any;
-    createMissingNativeApi():any;
-    initialized:boolean;
-    supported:boolean;
+    interface RangySelection extends Selection {
+        nativeSelection: Selection;
+
+        isBackwards(): boolean;
+
+        refresh(checkForChanges?: boolean): any;
+
+        toHtml(): string;
+
+        getAllRanges(): RangyRange[];
+
+        getRangeAt(idx: number): RangyRange;
+
+        getNativeTextRange(): any;
+
+        setSingleRange(range: RangyRange): any;
+
+        setRanges(ranges: RangyRange[]): any;
+
+        getBookmark(containerNode: Node): rangy.SelBookmark;
+
+        moveToBookmark(bookmark: rangy.Bookmark): void;
+
+        saveRanges(): object;
+
+        restoreRanges(saved: object): any;
+
+        saveCharacterRanges(containerNode: Node, opts?: any): object;
+
+        restoreCharacterRanges(containerNode: Node, characterRanges: object, opts?: any): any;
+
+        detach(): any;
+
+        inspect(): string;
+
+        move(units: string, count: number, opts?: any): number;
+    }
+
+    type RangyStatic = typeof Rangy;
 }
-declare module 'rangy' {
-    export = rangy;
+declare namespace Rangy {
+    function createNativeRange(doc?: Document | Window | HTMLIFrameElement): Range;
+
+    function createRange(doc?: Document | Window | HTMLIFrameElement): RangyRange;
+
+    function createRangyRange(doc?: Document | Window | HTMLIFrameElement): RangyRange;
+
+    function getNativeSelection(win?: Window): Selection;
+
+    function getSelection(doc?: Document | Window | HTMLIFrameElement): RangySelection;
+
+    function addInitListener(listener: (rangy: RangyStatic) => void): any;
+
+    function shim(): any;
+
+    function createMissingNativeApi(): any;
+
+    let initialized: boolean;
+    let supported: boolean;
+
+    interface Bookmark {
+        start: number;
+        end: number;
+        containerNode: Node;
+    }
+
+    interface SelBookmark {
+        backward: boolean;
+        rangeBookmarks: Bookmark[];
+    }
 }
-declare var rangy:RangyStatic;
+// we need export RangyStatic = typeof Rangy
+// tslint:disable-next-line:export-just-namespace
```

Comment:
Please remove this lint rule. Refer [this to write as module and global](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module) for more details.

Reply:
I can remove that lint rule because:
+ we need export RangyStatic = typeof Rangy (see the comment right above), so we need `declare namespace Rangy`
+ see also [my comment bellow](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34066#discussion_r268049871)